### ISSUE TITLE
Permalinks for IETF nightly URLs will lead to churn

### ISSUE THREAD

**Issue by foolip** [w3c/browser-specs#1127]
In https://github.com/web-platform-dx/web-features/pull/420 @ddbeck and I have run into a problem with the URLs for WebP added in https://github.com/w3c/browser-specs/pull/1120.

The nightly URL https://www.ietf.org/archive/id/draft-zern-webp-13.html has the version number (13) in the URL and will presumably keep changing. That will mean that projects that use browser-specs for URL validation (at least BCD and web-features) need to rename these URLs to match.

https://datatracker.ietf.org/doc/html/draft-zern-webp/ seems like a better nightly URL for this.

I'm filing an issue because I see at least one other case of this:
https://www.ietf.org/archive/id/draft-davidben-http-client-hint-reliability-03.html

**Comment by tidoust** [comment]
Right, see #1122 for a possible plan to make browser-specs get the data from the datatracker home page for these specs.

That said, the plan that I had in mind was more to have browser-specs detect these changes of URLs automatically, not necessarily to prevent them and switch to a versionless URL.

I'm fine switching to the `datatracker.ietf.org` URL instead of `www.ietf.org`. I just wonder whether the markup is significantly different, and whether one version could be seen as "better" than the other (see also #933 where `httpwg.org` versions are seen as more readable than `rfc-editor.org` ones).

**Comment by tidoust** [comment]
Some takeaways from a side discussion with @dontcallmedom about this today.

There are two types of URLs in browser-specs:
- Canonical URLs (the root `url` property), for which we favor stability over freshness and readability.
- Nightly URLs (the `nightly.url` property), for which we favor freshness and readability over stability.

For IETF specs that are not yet published as RFC, the `www.ietf.org` version seems to be the most readable HTML version of the spec for now (for example, the `datatracker.ietf.org` URL uses a monospace font, which makes it less readable than the `www.ietf.org` version), so we would prefer to continue using that URL as the nightly URL, even though that indeed creates data churn in practice because that URL contains a version number.

If web-features would rather use the `datatracker.ietf.org` URL in this case to avoid having to propagate the URL update, you could perhaps consider creating a conditional rule that says something like:

```js
const url = spec.nightly.url.match(/www\.ietf\.org\/archive/) ? spec.url : spec.nightly.url;
```

For what it's worth, neither property is guaranteed to be stable. I added some text on [spec identifiers](https://github.com/w3c/browser-specs/#spec-identifiers) to the README to clarify that.

**Comment by tidoust** [comment]
The code now detects releases of new versions for IETF drafts to `www.ietf.org`, and updates the nightly URL in browser-specs accordingly. As said above, that does not really solve your problem. You'll need to use the canonical URL instead. That is the "stable" URL of a spec.

I'm closing this issue for now. If updates prove painful over time, we could perhaps try to convince IETF to create a `https://www.ietf.org/archive/id/[draft name]-latest.html` URL for versioned drafts.
